### DISCUSSION TITLE
Error bed levelling

### DISCUSSION THREAD

**Discussion by pollosud** [xblax/flashforge_ad5m_klipper_mod#143] [General]
I have a problem with Z levelling.It gives this error  when I save the configuration
![IMG_20240412_185741-1](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/160148315/590390ad-085b-4eb1-81cf-13b2a2ef9d97)

**Comment by consp** [comment]
Can you tell if it also happens when you level via fluidd or mainsail?

**Comment by xblax** [comment]
Well I guess it's again the issue that dumb UI tools want to do `Z_OFFSET_APPLY_ENDSTOP` even though that makes no sense. Maybe the pragmatic approach avoid that would be to just overwrite the Macro and do `Z_OFFSET_APPLY_PROBE` instead.

**Reply by KaruroChori** [reply]
No, the likely issue is the separation between printer.base.cfg and printer.cfg.
Baseline klipper does not edit included files. I discovered it right few hours ago while updating the config structure of my flsun q5 to match the one of this repo, and trying to save my DELTA_CALIBRATE after :D.
There is a fix for that in https://github.com/DangerKlippers/danger-klipper, but it was never brought to main klipper.
There is no solution other than dropping the two file structure.

For reference: https://github.com/DangerKlippers/danger-klipper/pull/153

That being said, it is possible that for this printer Z_OFFSET_APPLY_PROBE is correct over Z_OFFSET_APPLY_ENDSTOP.

**Reply by xblax** [reply]
I think the separation between printer.base.cfg and printer.cfg is rather a feature than an issue in this case. Because Z_OFFSET_APPLY_ENDSTOP is not supposed to be used on that Printer and the GUI tools still do it. It's correct this fails with an error. If it was allowed to be saved that would cause only problems rather than actually adjusting the distance to the bed.
